### PR TITLE
[WIP] Scanner: skip files in submodules

### LINKED ISSUE #149: DataService: changes in a tree that is a sub-module
For now, to avoid failures as in #144 in DataService we just skip all requested files that are not part of the repository. One case when this is not an error is when the changes were done in some sub-module.

Rationale is: most probably that would not be a place user will want to see the feedback on his PR anyway.

Supporting getting files/changes from sub-modules with DataService _may_ make sense in some cases, so this issue is about:
 1. finding such cases
 2. if found, implementing sub-modules support
 3. if not found, may be we should not event detect these as a change and just skip inside `DiffTreeScanner`

### PR DESCRIPTION
**Pull request by bzz**
fixes #149 

 - [x] rollback #150 
 - [ ] tests for Tree/File scanners
 - [ ] a fix

CI is expected to fail

### CODE REVIEW

File: service/git/scanner.go [hunk 1]
```diff
@@ -12,7 +12,6 @@ import (
 	"gopkg.in/src-d/go-git.v4/plumbing/object"
```
Comment (aliciakozlowski) [inline service/git/scanner.go]: The file has increased in 1 lines.

File: service/git/scanner_test.go [hunk 2]
```diff
@@ -2,13 +2,16 @@ package git
```
Comment (aliciakozlowski) [inline service/git/scanner_test.go]: The file has increased in 65 lines.

File: service/git/scanner_test.go [hunk 3]
```diff
@@ -97,6 +104,31 @@ func (s *ScannerSuite) TestTreeScannerFiles() {
...
 }
 
+func (s *ScannerSuite) TestTreeScannerFilesSubmoduleUpdate() {
+	require := s.Require()
+	submodules := fixtures.ByTag("submodule").One()
+	fs := submodules.DotGit()
+	sto, err := filesystem.NewStorage(fs)
+	require.NoError(err)
+
+	//https://github.com/git-fixtures/submodule/compare/c7431b5bc9d45fb64a87d4a895ce3d1073c898d2...47770b26e71b0f69c0ecd494b1066f8d1da4fc03
```
Comment (aliciakozlowski) [inline service/git/scanner_test.go]: This line exceeded 120 chars.

File: service/git/scanner_test.go [hunk 4]
```diff
@@ -287,6 +319,39 @@ func main() {
 `, string(files["vendor/foo.go"].Content))
 }
 
+func (s *ScannerSuite) TestFileBlobScannerSubmoduleUpdate() {
+	require := s.Require()
+	//https://github.com/git-fixtures/submodule/compare/47770b26e71b0f69c0ecd494b1066f8d1da4fc03...b685400c1f9316f350965a5993d350bc746b0bf4
```
Comment (aliciakozlowski) [inline service/git/scanner_test.go]: This line exceeded 120 chars.

File: service/git/scanner.go (hunk 1 above)
Comment (juanjux) [inline service/git/scanner.go]: The file has increased in 1 lines.

File: service/git/scanner_test.go (hunk 2 above)
Comment (juanjux) [inline service/git/scanner_test.go]: The file has increased in 65 lines.

File: service/git/scanner_test.go (hunk 3 above)
Comment (juanjux) [inline service/git/scanner_test.go]: This line exceeded 120 chars.

File: service/git/scanner_test.go (hunk 4 above)
Comment (juanjux) [inline service/git/scanner_test.go]: This line exceeded 120 chars.

**Comment by juanjux** [comment]
Ups, sorry for the lookout comments, I was toying with it, maybe it shouldn't put this repo as default but a fake one, I'll write it into the friction log.

File: service/git/scanner.go (hunk 1 above)
Comment (jkblume) [inline service/git/scanner.go]: The file has increased in 1 lines.

File: service/git/scanner_test.go (hunk 2 above)
Comment (jkblume) [inline service/git/scanner_test.go]: The file has increased in 65 lines.

File: service/git/scanner_test.go (hunk 3 above)
Comment (jkblume) [inline service/git/scanner_test.go]: This line exceeded 120 chars.

File: service/git/scanner_test.go (hunk 4 above)
Comment (jkblume) [inline service/git/scanner_test.go]: This line exceeded 120 chars.

**Comment by carlosms** [comment]
Should we close this PR @bzz?

**Pull request closed without merging**